explorer/syntax/parser.ypp

```diff
@@ -104,11 +104,12 @@
 %type <std::vector<LibraryName>> import_directives
 %type <std::string> optional_library_path
 %type <bool> api_or_impl
+%type <bool> fn_virtual_override_intro
```

Comment:
Since we don't have `override` in Carbon, consider having this look more like "fn_virtual_modifier" or similar.

But you might also consider moving FN to function declaration, i.e.:

```
optional_virtual:
  // Empty
    { $$ = false; }
| VIRTUAL
    { $$ = true; }
;

function_declaration:
  optional_virtual FN ...
```

I think there'll be more keywords added to `fn` over time (e.g., public/private qualifiers), so doing keywords this way may make it easier to read overall.

Reply:
The above `optional_virtual` would also be reusable with destructors, versus the current approach which isn't.

Reply:
Right... so I maybe have anticipated the next step a bit here, without explaining :).
If you look in the next PR, I have to change the parser, and really, this is where this change came from (though I split both changes): https://github.com/carbon-language/carbon-lang/pull/2493/files#diff-f25d9773a1b66ce1fa19eb159c87c06cc7189fb2be419027c1401ce86c7cbdbbR1067

* I added an enum for that, and the other keywords needed, which of course, create a shift/reduce conflict due to `impl` for impls.
* Merging FN and the virtual override keyword eliminates the issue (though yes, TBD for destructors).
* "virtual_override" comes from it being the preferred term based on the design doc: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md#virtual-override-keywords

But I'm also relatively new to Bison, so it is possibly approach might be suboptimal.